components/brave_wallet/browser/zcash/zcash_wallet_service.h

```diff
@@ -150,9 +153,13 @@ class ZCashWalletService : public mojom::ZCashWalletService,
   friend class ZCashCreateShieldTransactionTask;
   friend class ZCashCreateTransparentTransactionTask;
   friend class ZCashDiscoverNextUnusedZCashAddressTask;
+  friend class ZCashResolveBalanceTask;
+  friend class ZCashShieldSyncService;
   friend class ZCashTransactionCompleteManager;
   friend class ZCashTxManager;
+
   friend class ZCashWalletServiceUnitTest;
+  friend class ZCashShieldSyncServiceTest;
```

Comment:
cc @cypt4 @yrliou 

I think there's a lot of room for improvement for how we are structuring access around `ZCashWalletService`, the data it consumes, and the data it provides. At the moment we have this:

`ZCashWalletService` holds several pointers, that it access internally as friend data. Those pointers also hold pointers back to `ZCashWalletService` and they in some cases also consume internal data from `ZCashWalletService`. This makes the whole data abstraction **porous**, and the rationale for it is not documented either in code, or in the comments, we are in [violations of the C++ Style Guide for the use of friend classes in Chromium[1]](https://google.github.io/styleguide/cppguide.html#Friends).

I think this needs to be corrected, to a type of abstraction that encapsulates data more consistently, and I've created an [issue](https://github.com/brave/brave-browser/issues/42022) for this.

[1] https://google.github.io/styleguide/cppguide.html#Friends
[2] https://github.com/brave/brave-browser/issues/42022